### ISSUE TITLE
Integration of jsonata-moment for support of correct timely based conditions in summer and winter time

### ISSUE THREAD

**Issue by Adrian-Grimm** [louislam/uptime-kuma#5016]
### 📑 I have found these related issues/pull requests

-

### 🏷️ Feature Request Type

Other

### 🔖 Feature description

Some JSON validations require a time based condition e.g. device states with time based states.
But in countries like Germany we have summer and winter time difference in offsets and the default $now() only supports a fixed timezone offset. 

### ✔️ Solution

Integration of jsonata-moment for support of correct timely based conditions in summer and winter time with $moment()
https://github.com/elasticio/jsonata-moment

**Comment by CommanderStorm** [comment]
I'd like to not add jet another datetime library.
I would prefer if jsonata would get the functionality to localize dates upstream instead.

**Comment by CommanderStorm** [comment]
Yes, it has been a few years, but without anybody contributing a PR.
If someone contributes a PR upstream, I'd assume that they will review it.

As said:
I'd prefer if you'd contribute the functionality you need upstream.
we can add the plugin if upstream
- rejects your PR or
- comments that they'd like to keep it in a separate plugin

Monkeypatching functionality without a reason is not maintainable for us, sorry.

**Comment by Adrian-Grimm** [comment]
@CommanderStorm :
Fyi: Seems like a badly maintained project - I got no reply since August there:
https://github.com/jsonata-js/jsonata/issues/347#issuecomment-2293647619

**Comment by CommanderStorm** [comment]
Consider opening up a PR instead. Commenting on an issue is something that requires a lot more effort to review, merge, ...

Also about the timing: once that notification gets clicked away as "I don't currently have the time" it does not reappear for maintainers. PRs are simpler to review.
